### ISSUE TITLE
A numerical value of 0 is incorrectly treated as falsy

### ISSUE THREAD

**Issue by latimer1** [jg-rp/liquid#74]
Per https://shopify.github.io/liquid/basics/truthy-and-falsy/ "All values in Liquid are truthy except nil and false."

However, the integer `0` and float `0.0` are being treated as falsy.

Trying the following code input:
```
{% if 0.0 %} 0.0 float {% endif %}
{% if 0 %} 0 int {% endif %}
{% if '0' %} 0 string {% endif %}
```

Output:
```
0 string
```

If this is intended, I don't see this boolean behavior defined in known issues in https://jg-rp.github.io/liquid/known_issues

**Comment by latimer1** [comment]
The issue is https://github.com/jg-rp/liquid/blob/main/liquid/expression.py#L914

 obj in (False, None) evaluates to true because `0.0 == False` is True.

**Comment by jg-rp** [comment]
Hi @latimer1,

I believe the case of `{% if 0 %}...` was fixed in version 1.4.2 with #65, but I hadn't considered `{% if 0.0 %}...`, so that's defiantly a bug that needs fixing.

Expected output from your example:
```
 0.0 float 
 0 int 
 0 string
```

Thanks for letting me know.

**Comment by latimer1** [comment]
I wonder if adding a type check would address any edge cases.
```
if type(obj) == type(False) and obj == False:
   return False
if type(obj) == type(None) and obj == None:
   return False
```

**Comment by jg-rp** [comment]
There are a couple of cases that that does not cover. [Undefined variables](https://jg-rp.github.io/liquid/guides/undefined-variables) and [drops](https://jg-rp.github.io/liquid/introduction/objects-and-drops) that define `__bool__` or `__int__` methods can be falsy too.

This bug also extends to comparison operators and the `default` filter:
```liquid
{% if 0.0 == false %}0.0 == false{% endif %}
{{ 0.0 | default: "hello" }}
```

**Expected output**
```plain

0.0
```

**Actual output**
```plain
0.0 == false 
hello
```

**Comment by jg-rp** [comment]
Fixed in version 1.4.5, now released.
